vagrant/provisioning/roles/solr/templates/solrconfig.xml

```diff
@@ -35,7 +35,7 @@
        that you fully re-index after changing this setting as it can
        affect both how text is indexed and queried.
   -->
-  <luceneMatchVersion>7.7.2</luceneMatchVersion>
+  <luceneMatchVersion>${lucene_Match_Version}</luceneMatchVersion>
```

Comment:
Our facts should be all lower case: `lucene_match_version`.  

Also, if you provide a default value, we don't have to update any facts files: `{{ lucene_match_version | default(solr_version) }}`.   Also, notice here the correct Ansible template syntax, it should be `{{ some_config_item_name }}`, versus `${ some_config_item_name }`